lib/administration/zone/include/irods/zone_administration.hpp

```diff
@@ -0,0 +1,287 @@
+
+/// \file
+
+#undef NAMESPACE_IMPL
+#undef RxComm
+#undef rxGeneralAdmin
+
+#ifdef IRODS_ZONE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+    // NOLINTBEGIN(cppcoreguidelines-macro-usage)
+    #define NAMESPACE_IMPL server
+    #define RxComm         RsComm
+    #define rxGeneralAdmin rsGeneralAdmin
+    #define IRODS_QUERY_ENABLE_SERVER_SIDE_API
+    // NOLINTEND(cppcoreguidelines-macro-usage)
+
+    #include "irods/rodsConnect.h"
+    #include "irods/rsGeneralAdmin.hpp"
+
+struct RsComm;
+#else
+    // NOLINTBEGIN(cppcoreguidelines-macro-usage)
+    #define NAMESPACE_IMPL client
+    #define RxComm         RcComm
+    #define rxGeneralAdmin rcGeneralAdmin
+    // NOLINTEND(cppcoreguidelines-macro-usage)
+    
+    #include "irods/generalAdmin.h"
+
+struct RcComm;
+#endif // IRODS_ZONE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include "irods/irods_exception.hpp"
+#include "irods/irods_query.hpp"
+#include "irods/rodsErrorTable.h"
+#include "irods/zone_type.hpp"
+
+#include <fmt/format.h>
+
+#include <concepts>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+namespace irods::experimental::administration
+{
+    /// \since 4.3.1
+    inline namespace v1
+    {
+        /// Represents the set of valid ACL values for a zone collection.
+        ///
+        /// \since 4.3.1
+        enum class zone_collection_acl
+        {
+            null,
+            read
+        }; // enum class zone_collection_acl
+
+        /// Holds the name of a zone. Primarily used to modify a zone.
+        ///
+        /// \since 4.3.1
+        struct zone_name_property
+        {
+            std::string value;
+        }; // struct zone_name_property
+
+        /// Holds the connection information of a zone. Primarily used to modify a zone.
+        ///
+        /// \since 4.3.1
+        struct connection_info_property
+        {
+            std::string value;
+        }; // struct connection_info_property
+
+        /// Holds the comment attached to a zone. Primarily used to modify a zone.
+        ///
+        /// \since 4.3.1
+        struct comment_property
+        {
+            std::string value;
+        }; // struct comment_property
+
+        /// Holds the ACL and entity to attach to a zone. Primarily used to modify a zone.
+        ///
+        /// \since 4.3.1
+        struct zone_collection_acl_property
+        {
+            // clang-format off
+            zone_collection_acl acl; ///< The permission for a zone collection.
+            std::string name;        ///< The name of a user or group.
+            // clang-format on
+        }; // struct zone_collection_acl_property
+
+        // clang-format off
+        template <typename T>
+        concept ZoneProperty = std::same_as<T, zone_name_property> ||
+            std::same_as<T, connection_info_property> ||
+            std::same_as<T, comment_property> ||
+            std::same_as<T, zone_collection_acl_property>;
```

Comment:
Doxygen does not understand concepts.

Leave a normal comment or see if there's a way to make Doxygen produce documentation for it.

Reply:
can bump Doxygen if necessary without too much trouble.

Reply:
Verified that doxygen can handle concepts just fine. No need to bump anything.